fairscale/experimental/nn/multiprocess_pipe.py

```diff
@@ -0,0 +1,198 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+from collections import OrderedDict
+import itertools
+from typing import TYPE_CHECKING, Callable, Dict, List, Tuple, Type, Union
+
+import torch
+from torch import Tensor
+import torch.distributed.rpc as rpc
+import torch.nn as nn
+
+Tensors = Tuple[Tensor, ...]
+TensorOrTensors = Union[Tensor, Tensors]
+
+LayerSpec = Tuple[str, Type[nn.Module], Tuple, Dict]
+
+if TYPE_CHECKING:
+    Module = nn.Module[TensorOrTensors]
+else:
+    Module = nn.Module
+
+BOUNCE_TENSORS = True
+
+
+def _verify_module(module: List[LayerSpec]) -> None:
+    if not isinstance(module, List):
+        raise TypeError("module must be a list")
+
+    for elem in module:
+        if not isinstance(elem, tuple):
+            raise TypeError("module must be a list of tuple")
```

Comment:
Should we say "elem must be of ..."?

Reply:
Also, does it make sense to print out what was passed in as part of the error message? just for easier debugging.

Reply:
module is correct in this context. module is the name constructor param. The error is telling the API user they passed in a value of the wrong type.

Reply:
> Also, does it make sense to print out what was passed in as part of the error message? just for easier debugging.

I usually like to print things out but in this case is a constructor parameter so it should fail pretty quickly. If it fails, it essentially means you are using the API incorrectly.